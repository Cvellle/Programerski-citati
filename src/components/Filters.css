.filters-inner {
  position:fixed;
  top: 0;
  bottom:0;
  overflow-y:scroll;
  overflow-x:hidden;
  padding-right: 2%;
  width: inherit;
}

.filters-inner div {
  cursor: pointer;
  margin: 8px 0;
}

.filters-inner div:nth-child(odd) {
  background: #e5e5e5;
}

.filters-inner div:nth-child(even) {
  background: #f5f5f5;
}

.filters-inner img {
  margin-right: 8px;
  width: 30px;
}

.filters-inner input,
.filters-inner button {
  font-size: inherit;
}

.filters-inner input {
  width: 100%;
}

.svi-autori {
  margin-bottom: 20px;
}
