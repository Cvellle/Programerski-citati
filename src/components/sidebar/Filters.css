.filters input,
.filters button {
  font-size: inherit;
}

.filters input {
  width: 100%;
}

.btn-authors {
  margin-bottom: 20px;
}
