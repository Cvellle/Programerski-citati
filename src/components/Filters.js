import React from 'react';
import {sortirajAbecedno} from '../shared/helpers'
import './Filters.css'

const Filters = props => {

  const sortirano = [...props.autori].sort(sortirajAbecedno)

  const autori = sortirano.map((autor, i) =>
    <div key={i} onClick={() =>props.izaberiAutora(autor)}>
      {props.slikeAutora.get(autor) ? <img src={props.slikeAutora.get(autor)} alt={autor} /> : ''}
      {autor}
    </div>
  )

  return (
    <aside className="filters">
      <div className="filters-inner">

        <h3>{props.jezik === 'en' ? 'Search text' : 'Pretra≈æi tekst'}</h3>
        <input name="tekst" onChange={props.setPhrase} />

        <h3>{props.jezik === 'en' ? 'Choose the author' : 'Izaberi autora'}</h3>
        <button className="svi-autori" onClick={() =>props.izaberiAutora('')}>{props.jezik === 'en' ? 'All authors' : 'Svi autori'}</button>
        {autori}

      </div>
    </aside>
  )
}

export default Filters
