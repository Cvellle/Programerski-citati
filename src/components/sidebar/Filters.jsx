import React from 'react'
import {Link} from 'react-router-dom'
import './Filters.css'
import translate from '../../shared/translate'

const Filters = ({ setPhrase, filterAuthors }) => (
  <div className="filters">
    <h3>{translate('SEARCH_TEXT')}</h3>
    <Link className="author" to="/"><input onChange={setPhrase} /></Link>

    <h3>{translate('CHOOSE_AUTHOR')}</h3>
    <input placeholder={translate('SEARCH_FOR_AUTORS')} onChange={(e) => filterAuthors(e.target.value)} />
  </div>
)

export default Filters
