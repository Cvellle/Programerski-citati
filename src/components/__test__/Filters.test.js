// @flow
import React from 'react';
import { shallow, mount } from 'enzyme';
import Filters from '../Filters';

const setPhrase = jest.fn();
const setAuthor = jest.fn();

describe('Component: Filters snapshot', () => {
    it('should render Filters ', () => {
        const wrapper = shallow(<Filters authorImages={new Map()} authors={["some","random", "authors"]} language="en" setPhrase={() => {}} setAuthor={() => {}} />);

        expect(wrapper).toMatchSnapshot();
    });
});

describe('Filters component', () => {
    const wrapper = shallow(
      <Filters
        authorImages={new Map()}
        authors={["some","random", "authors"]}
        language="en"
        setPhrase={setPhrase}
        setAuthor={setAuthor}
      />
    );

    const language = wrapper.find("[name='language']");
    const phraseInput = wrapper.find("input");
    const button = wrapper.find("button");

    test('it renders filters language', () => {
        expect(language.exists()).toBeTruthy();
    });

    test('it renders phraseInput language and call setPhrase', () => {
        expect(phraseInput.exists()).toBeTruthy();
        phraseInput.simulate('change');
        expect(setPhrase).toBeCalled();
    });

    test('it renders allAuthors button and call setAuthor', () => {
        expect(button.exists()).toBeTruthy();
        button.simulate('click');
        expect(setAuthor).toBeCalled();
    });
});
