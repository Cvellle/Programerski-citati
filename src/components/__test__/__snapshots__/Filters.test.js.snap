// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Filters snapshot should render Filters  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filters
    authorImages={Map {}}
    authors={
        Array [
            "some",
            "random",
            "authors",
          ]
    }
    language="en"
    setAuthor={[Function]}
    setPhrase={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="filters-inner"
>
        <h3
                name="language"
        >
                Search text
        </h3>
        <input
                name="tekst"
                onChange={[Function]}
        />
        <h3>
                Choose the author
        </h3>
        <button
                className="svi-autori"
                onClick={[Function]}
        >
                All authors
        </button>
        <div
                onClick={[Function]}
        >
                
                authors
        </div>
        <div
                onClick={[Function]}
        >
                
                random
        </div>
        <div
                onClick={[Function]}
        >
                
                some
        </div>
</div>,
      "className": "filters",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            name="language"
>
            Search text
</h3>,
          <input
            name="tekst"
            onChange={[Function]}
/>,
          <h3>
            Choose the author
</h3>,
          <button
            className="svi-autori"
            onClick={[Function]}
>
            All authors
</button>,
          Array [
            <div
              onClick={[Function]}
>
              
              authors
</div>,
            <div
              onClick={[Function]}
>
              
              random
</div>,
            <div
              onClick={[Function]}
>
              
              some
</div>,
          ],
        ],
        "className": "filters-inner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search text",
            "name": "language",
          },
          "ref": null,
          "rendered": "Search text",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "tekst",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Choose the author",
          },
          "ref": null,
          "rendered": "Choose the author",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "All authors",
            "className": "svi-autori",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "All authors",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              "authors",
            ],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "",
            "authors",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              "random",
            ],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "",
            "random",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              "some",
            ],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "",
            "some",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="filters-inner"
>
          <h3
                    name="language"
          >
                    Search text
          </h3>
          <input
                    name="tekst"
                    onChange={[Function]}
          />
          <h3>
                    Choose the author
          </h3>
          <button
                    className="svi-autori"
                    onClick={[Function]}
          >
                    All authors
          </button>
          <div
                    onClick={[Function]}
          >
                    
                    authors
          </div>
          <div
                    onClick={[Function]}
          >
                    
                    random
          </div>
          <div
                    onClick={[Function]}
          >
                    
                    some
          </div>
</div>,
        "className": "filters",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              name="language"
>
              Search text
</h3>,
            <input
              name="tekst"
              onChange={[Function]}
/>,
            <h3>
              Choose the author
</h3>,
            <button
              className="svi-autori"
              onClick={[Function]}
>
              All authors
</button>,
            Array [
              <div
                onClick={[Function]}
>
                
                authors
</div>,
              <div
                onClick={[Function]}
>
                
                random
</div>,
              <div
                onClick={[Function]}
>
                
                some
</div>,
            ],
          ],
          "className": "filters-inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search text",
              "name": "language",
            },
            "ref": null,
            "rendered": "Search text",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "tekst",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Choose the author",
            },
            "ref": null,
            "rendered": "Choose the author",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "All authors",
              "className": "svi-autori",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "All authors",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                "authors",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "",
              "authors",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                "random",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "",
              "random",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                "some",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "",
              "some",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
